src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserGroupController.kt

```diff
@@ -134,4 +135,59 @@ class UserGroupController(
     userGroupService.removeGroupByUserId(userId, group)
     log.info("Remove group succeeded for userId {} and group {}", userId, group)
   }
+
+  @PutMapping("/users/{userId}/groups/{group}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
+    summary = "Add group to user.",
+    description = "Add group to user."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Added"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "User not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Group for user already exists.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun addGroupByUserId(
+    @Parameter(description = "The userId of the user.", required = true) @PathVariable
+    userId: UUID,
+    @Parameter(description = "The group to be added to the user.", required = true) @PathVariable
```

Comment:
Minor - the **group code** of the group to be added to the user

Reply:
Done